File: 0049_Group Anagrams/GroupAnagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var dict = new Dictionary<string, IList<string>>();
        
        foreach(var s in strs)
        {
            var s_arr = s.ToCharArray();
            Array.Sort(s_arr);
            var new_s = new string(s_arr);
            if(!dict.ContainsKey(new_s)) dict.Add(new_s, new List<string>());
            dict[new_s].Add(s);
        }
        
        return dict.Values.ToList();
    }
}

File: 0049_Group Anagrams/GroupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        
        for(String s : strs) {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String newStr = new String(chars);
            map.putIfAbsent(newStr, new ArrayList<String>());
            map.get(newStr).add(s);
        }
        
        List<List<String>> ans = new ArrayList<>();
        for(List<String> list : map.values())
            ans.add(list);
        
        return ans;
    }
}